### ISSUE TITLE
Allow to train policy on actual user messages

### ISSUE THREAD

**Issue by tmbo** [RasaHQ/rasa_core#287]
Right now the policy is trained on intents, the labels assigned to the incoming messages from NLU.

There are models we want to try out that would benefit from the additional information the whole message provides. The training data passed into the Rasa Core policy should contain the nlu user message.

**Comment by amn41** [comment]
I wonder if we should have an `UnfeaturizedTrainingData` class that holds everything as tuples, labels, which then gets pass to the `Featurizer` to turn it into `X, y`

**Comment by ctrado18** [comment]
I wonder if you can write stories also without any intents? What about if I just want to use NER and want to have stories with actions after recognized entities and no intents at all.
Altough I have the feeling that you are ""trapped" with intents, is it possible to be more flexible? Lets say you are just interested if the intent right now is anything else than a specific intent in the sate of conversation? This I want to use for fall-back strategies.

**Comment by amn41** [comment]
yes we're currently experimenting with how robustly this works!

**Comment by ctrado18** [comment]
Ok that would be amazing! But I want to check if I am correct. So far, it is possible to write such custom behaviour (like check if intent is anything else than a specific intent after a specific ation or intent) in the policy's probabilities_using_best_policy function. There I can execute a specific action for my situations with this return value like:

`return utils.one_hot(action, domain.num_actions)` 

Am I right?

Doing such things in a custom action is not possible or not a good way? Like executing a custom action inside a custom action? Still, I am asking myself how one cann call an action (like the restart action) inside a custom action?

That would be great if you can shortly reply.

**Comment by andreasstuber** [comment]
The context of a chat often involves the last exchanged messages as well, not only the current message. Have you thought about doing a CNN that looks at the last X messages? Possibly one that gives greater weights to closer messages to predict next actions? Or doing a tensorflow_embedding across the last x messages (and not only the current message). 
An additional idea would be to also include in this CNN the utterances of the bot (which can be highly condensed unlike unstructured user chats, so adding bot msgs to this CNN wouldn't add a lot of dimensionality)  - what a user says is often a response to what a bot said.

**Comment by Ghostvv** [comment]
@andreasstuber do you speak about NLU or core? In rasa_core we use RNN for the functionality you describe
